src/eval.rs

```diff
@@ -665,6 +665,29 @@ pub fn eval_full<R>(
     global_env: &Environment,
     resolver: &mut R,
 ) -> Result<RichTerm, EvalError>
+where
+    R: ImportResolver,
+{
+    eval_internal(t0, global_env, resolver).map(|(term, env)| subst(term, global_env, &env))
+}
+
+/// Fully evaluates a Nickel term like `eval_full`, but does not substitute all variables.
+pub fn eval_deep<R>(
+    t0: RichTerm,
+    global_env: &Environment,
+    resolver: &mut R,
+) -> Result<RichTerm, EvalError>
+where
+    R: ImportResolver,
+{
+    eval_internal(t0, global_env, resolver).map(|(term, _)| term)
+}
+
+fn eval_internal<R>(
```

Comment:
Nitpick: `internal` is not very suggestive. Maybe `eval_deep_` or `eval_deep_closure` or something.